### PR TITLE
Create Redirector_Setup

### PR DESCRIPTION
**Pull request by MintMain21**
Guidelines on setting up Redirector to act as a YouTube-Invidious extension.

### CODE REVIEW

**Review by MintMain21 — COMMENTED** [review]
It just occurred to me that additional rules need to be added for the https://m.youtube.com/ domain, for both video redirects with URL Parameters and general YouTube redirects (channels, homepage, etc).

File: docs/Redirector_Setup [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# What is Redirector?
+
+Redirector is a Browser extension for Firefox and Chromium that can manually be configured to redirect links, such as YouTube to Invidious or Twitter to Nitter (https://github.com/zedeus/nitter/wiki/Extensions#redirector)
+
+# How to use it for Invidious Redirects?
+
+For basic redirects, you will need the following rules set up.
+
+#Rule 1
```
Comment (TheFrenchGhosty) [inline docs/Redirector_Setup:9]: You need a space between `#`and the text.

File: docs/Redirector_Setup [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+Redirects Youtube Videos to Invidious
+Redirect: https://*.youtube.com/watch?*v=*
+to: https://[Invidious Domain Here]/watch?$2v=$3
+Hint: Redirects individual videos to their Invidious equivelant
+Example: https://www.youtube.com/watch?v=Lz13cxZNTCo → https://[Invidious Domain Here]/watch?v=Lz13cxZNTCo
+Applies to: Main window (address bar)
+
+#Rule 2 (Note, Rule 2 is for redirecting the youtu.be domain, typically accociated with embeds)
```
Comment (TheFrenchGhosty) [inline docs/Redirector_Setup:18]: ditto

File: docs/Redirector_Setup [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+Redirect: https://youtu.be/*
+to: https://[Invidious Domain Here]/watch?v=$1
+Example: https://youtu.be/Ae5yV4p7uh8?t=35 → https://[Invidious Domain Here]/watch?v=Ae5yV4p7uh8?t=35
+
+#Rule 3 (Note, Rule 3 is for those who use DDG bangs or search extensions to search YouTube and would like those properly redirected to Invidious)
```
Comment (TheFrenchGhosty) [inline docs/Redirector_Setup:24]: ditto

File: docs/Redirector_Setup [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+Redirect YouTube Search to Invidious
+Redirect: https://www.youtube.com/results?search_query=*&page
+to: https://[Invidious Domain Here]/search?q=$1
+Example: https://www.youtube.com/results?search_query=Smash+Bros+Trailer&page → https://[Invidious Domain Here]/search?q=Smash+Bros+Trailer
+Applies to: Main window (address bar)
+
+
+#Rule 4
```
Comment (TheFrenchGhosty) [inline docs/Redirector_Setup:33]: ditto

File: docs/Redirector_Setup [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+Applies to: Main window (address bar)
+
+
+If you would like to use URL Parameters in your redirects (see https://github.com/cloudrac3r/invidious-documentation/blob/master/List-of-URL-parameters.md), you will need to manually add them to Rule 1, (https://[Invidious Domain Here]/watch?v=$1[URL Parameters here] in place of https://[Invidious Domain Here]/watch?v=$1.
+
+For clarity, an example of a URL parameter would be &related_videos=false&comments=false to ensure that neither related videos nor comments are displayed on a video page, so https://[Invidious Domain Here]/watch?v=$1&related_videos=false&comments=false would be the destination of redirects.
+
+Should you choose to use URL Parameters, I would reccomend adding the following rule to both ensure all invidious videos load with your preference, and to ensure that unnecesary parameters are not added when already applied. 
+
+#Rule 5
```
Comment (TheFrenchGhosty) [inline docs/Redirector_Setup:48]: ditto

File: docs/Redirector_Setup [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+excluding: https://[Invidious Domain Here]/watch?v=*[URL Parameters here]
+Hint: Update this to correlate with the YouTube Video Redirect
+Example: https://[Invidious Domain Here]/watch?v=Lz13cxZNTCo → https://[Invidious Domain Here]/watch?v=Lz13cxZNTCo[URL Parameters here]
+Applies to: Main window (address bar)
+
+It is important that you add the exception so as not to create infinite repeats of the same URL Parameters when opening multiple Invidious Videos in a single session. When creating or editing rules in Redirector you must access "Advanced Options" in order to set exceptions. 
+
+For redirecting embeds, you'll need the following two rules.
+
+#Rule 6
```
Comment (TheFrenchGhosty) [inline docs/Redirector_Setup:62]: ditto

File: docs/Redirector_Setup [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+Redirects Embedded Youtube Videos to Invidious
+Redirect: https://www.youtube.com/embed/*
+to: https://iteroni.com/embed/$1
+Hint: Redirects individual videos to their Invidious equivelant
+Example: https://www.youtube.com/embed/Lz13cxZNTCo → https://iteroni.com/embed/Lz13cxZNTCo
+Applies to: Main window (address bar), IFrames
+
+#Rule 7
```
Comment (TheFrenchGhosty) [inline docs/Redirector_Setup:71]: ditto

File: docs/Redirector_Setup [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+Hint: Redirects individual videos to their Invidious equivelant
+Example: https://www.youtube-nocookie.com/embed/Lz13cxZNTCo → https://iteroni.com/embed/Lz13cxZNTCo
+Applies to: Main window (address bar), IFrames
+
+(Note, it is imparative that you check it applies to both the Main Window and IFrames, otherwise the rule won't applied to Embeds).
+
+
+Finally, if you like to subscribe to channels using RSS feeds, you made add the following two rules to redirect Invidious RSS feeds to their YouTube equivelants. This way you'll still have access to your feeds should you need to migrate to another instance. 
+
+#Rule 8
```
Comment (TheFrenchGhosty) [inline docs/Redirector_Setup:85]: ditto

File: docs/Redirector_Setup [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+Redirect Invidious Feeds to YouTube Feeds
+Redirect: https://[Invidious Domain Here]/feed/channel/*
+to: https://www.youtube.com/feeds/videos.xml?channel_id=$1
+Hint: And now you can subscribe in peace.
+Example: https://[Invidious Domain Here]/feed/channel/UCOkL7q2SeGZeZuj22njMYEA → https://www.youtube.com/feeds/videos.xml?channel_id=UCOkL7q2SeGZeZuj22njMYEA
+Applies to: Main window (address bar)
+
+#Rule 9
```
Comment (TheFrenchGhosty) [inline docs/Redirector_Setup:94]: ditto

**Review by TheFrenchGhosty — CHANGES_REQUESTED** [review]
Please use only lower-case letters in the file name + add .md

Don't use iteroni as an example instance, use a placeholder ideally `<url>`.

**Comment by MintMain21** [comment]
I think I addressed everything.

And I apologize for leaving the Iteroni link without replacing the domain.

File: docs/redirector_setup.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
+# What is Redirector?
+
+Redirector is a Browser extension for Firefox and Chromium that can manually be configured to redirect links, such as YouTube to Invidious or Twitter to Nitter (https://github.com/zedeus/nitter/wiki/Extensions#redirector)
```
Comment (TheFrenchGhosty) [inline docs/redirector_setup.md:3]:
```suggestion
Redirector is a Browser extension for Firefox and Chromium that can manually be configured to redirect links, such as YouTube to Invidious or [Twitter to Nitter](https://github.com/zedeus/nitter/wiki/Extensions#redirector)

Get redirector: [Firefox](https://addons.mozilla.org/en-US/firefox/addon/redirector/) — [Chromium-based](https://chrome.google.com/webstore/detail/redirector/ocgpenflpmgnfapjedencafcfakcekcd) — [GitHub](https://github.com/einaregilsson/Redirector)
```

**Comment by TheFrenchGhosty** [comment]
I pushed some changes myself, all good now, thank you for your contribution.

**Review by MintMain21 — COMMENTED** [review]
Hang on, I've noticed two mistakes. 

1. In Rule 4, the asterisk in the youtube domain (*.youtube.com) isn't showing.

2. Rule 9 isn't spaced far enough from the hashtag.

Don't know if this was on my end or yours, but it seems too late for me to change it without creating a new branch.

**Pull request merged**